scrapy/spiders/crawl.py

```diff
@@ -128,6 +140,19 @@ async def _parse_response(self, response, callback, cb_kwargs, follow=True):
             for request_or_item in self._requests_to_follow(response):
                 yield request_or_item
 
+    def parse_with_rules(self, response, callback, cb_kwargs, follow=True):
+        return self._parse_with_rules(response, callback, cb_kwargs, follow)
+
+    def _parse_response(self, response, callback, cb_kwargs, follow=True):
+        if not self._is_warned:
+            self._is_warned = True
```

Comment:
I think we need to distinguish warning from overriding and warning from use.

If you override, you get 1 warning. If you use, you get 1 warning. If you do both, you get 2 warnings.